spacewalk/setup/bin/spacewalk-setup-cobbler

```diff
@@ -53,10 +53,15 @@ def manipulate_cobbler_settings(config_dir: str, settings_yaml: str):
     backup_file(full_path)
     with open(full_path) as settings_file:
         filecontent = yaml.safe_load(settings_file.read())
+
     filecontent["server"] = socket.getfqdn()
-    filecontent["next_server_v4"] = socket.getaddrinfo(filecontent["server"], None, socket.AF_INET)[1][4][0]
 
-    # In case there is no IPv6 address, we get a OSError (socket.gaierror)
+    # In case of failing DNS resolution, we get a OSError (socket.gaierror)
+    try:
+        filecontent["next_server_v4"] = socket.getaddrinfo(filecontent["server"], None, socket.AF_INET)[1][4][0]
+    except OSError:
+        filecontent["next_server_v4"] = ""
```

Comment:
small question, what happens when "next_server_v4" and "next_server_v6" are both empty? Is any functionality impacted?

Reply:
Without the next server Cobbler is not functional. No auto-installations can work. An IP that the clients can resolve must be present.

Reply:
These are new settings in Cobbler 3.3.3, after looking at Cobbler code I see both of them are accepted as empty values.

In principle, as having DNS resolution working is a premise, you should get at least IPv4 address resolution. I added here the "try/except" because while debugging, using wrong hostname in "server", I noticed this would make the execution of `spacewalk-setup-cobbler` to crash ending up in a more inconsistent state.

Reply:
If I understand @SchoolGuy correctly, it doesn't make sense to continue with the script if the "server" hostname is not resolvable.

IMO we should check if we have at least one "next server" IP before this function overrides the old settings. Silently creating a broken (as in "not working") config is more inconsistent than bailing out.

Reply:
Well if we could raise the problem to the UI that we have a broken config, all daemons would start and we could use the Cobbler API to fix that during runtime. However that is sadly not possible afaik and thus I agree with @agraul that we should bail out and print a clear message with the reasoning for bailing out.

Reply:
Ok, I have made the script to crash in case we cannot resolve any IP addresses for the fqdn.

Also, in case any exception during the migration, the error is shown and execution will stop. Zypper displays a warning meesage and will not exit cleanly. Making the user hopefully to notice that something when wrong.

Triggering reinstallation of `spacewalk-setup` package should make the Cobbler migration to run again.